docs/design/configs.md

```diff
@@ -0,0 +1,231 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores command-line flags, environment variables, and configuration files used 
```

Comment:
I'm not sure we need to specify "command-line flags, environment variables, and configuration files". It's probably sufficient just to say "information used for the configuration of Kubernetes components".